sdk/python/kfp/cli/components.py

```diff
@@ -0,0 +1,382 @@
+        with open(filepath, 'w') as f:
+            f.write('# Generated by KFP.\n{}'.format(contents))
+        _info('Generated file {}.'.format(filepath))
+
+    def maybe_generate_requirements_txt(self):
+        self._maybe_write_file(_REQUIREMENTS_TXT, '')
+
+    def maybe_generate_dockerignore(self):
+        self._maybe_write_file(_DOCKERIGNORE, _DOCKERIGNORE_TEMPLATE)
+
+    def maybe_generate_dockerfile(self, overwrite_dockerfile: bool = False):
+        dockerfile_contents = _DOCKERFILE_TEMPLATE.format(
+            base_image=self._base_image,
+            maybe_copy_kfp_package=self._maybe_copy_kfp_package,
+            component_root_dir=component_factory.COMPONENT_ROOT_DIR,
+            kfp_package_path=self._kfp_package_path)
+
+        self._maybe_write_file(_DOCKERFILE, dockerfile_contents,
+                               overwrite_dockerfile)
+
+    def build_image(self, skip_pushing_image: bool = False):
+        _info('Building image {} using Docker...'.format(
+            typer.style(self._target_image, fg=typer.colors.YELLOW)))
+        client = docker.from_env()
+
+        docker_log_prefix = typer.style('Docker', fg=typer.colors.CYAN)
+
+        try:
+            context = str(self._context_directory)
+            _, logs = client.images.build(path=context,
+                                          dockerfile='Dockerfile',
+                                          tag=self._target_image)
+            for log in logs:
+                message = log.get('stream', '').rstrip('\n')
+                if message:
+                    _info('{}: {}'.format(docker_log_prefix, message))
+
+        except docker.errors.BuildError as e:
+            for log in e.build_log:
+                message = log.get('message', '').rstrip('\n')
+                if message:
+                    _error('{}: {}'.format(docker_log_prefix, message))
+            _error('{}: {}'.format(docker_log_prefix, e))
+            raise typer.Exit(1)
+
+        if skip_pushing_image:
+            return
+
+        _info('Pushing image {}...'.format(
+            typer.style(self._target_image, fg=typer.colors.YELLOW)))
+
+        try:
+            response = client.images.push(self._target_image,
+                                          stream=True,
+                                          decode=True)
+            for log in response:
+                status = log.get('status', '').rstrip('\n')
+                layer = log.get('id', '')
+                if status:
+                    _info('{}: {} {}'.format(docker_log_prefix, layer, status))
+        except docker.errors.BuildError as e:
+            _error('{}: {}'.format(docker_log_prefix, e))
+            raise e
+
+        _info('Built and pushed component container {}'.format(
+            typer.style(self._target_image, fg=typer.colors.YELLOW)))
+
+
+@app.callback()
+def components():
+    """Builds shareable, containerized components."""
+    pass
```

Comment:
nit: do we need this at some point?

Reply:
Yes, this is actually the main typer command group. See https://typer.tiangolo.com/tutorial/commands/callback/